clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -233,7 +233,11 @@ export default class ApiService {
 
     this.dataService.rotateAssignmentList(assignment);
 
-    const data = { ...assignment, assignedCondition: { ...assignment.assignedCondition[0], conditionCode: condition } };
+    const data = {
+      ...assignment,
+      assignedCondition: { ...assignment.assignedCondition[0], conditionCode: condition },
+      assignedFactor: assignment.assignedFactor?.[0] ? { ...assignment.assignedFactor?.[0] } : undefined,
+    };
```

Comment:
`assignedFactor: assignment.assignedFactor?.[0] ? { ...assignment.assignedFactor?.[0] } : undefined,` is the same as `assignedFactor: assignment.assignedFactor?.[0]`, ternary should be unnecessary

Reply:
code updated